### PR TITLE
Changed transaction sender validation

### PR DESCRIPTION
**Pull request by razvantomegea**
### Feature
Allow sign transactions when logged in with the sender's active guardian address

### CODE REVIEW

File: src/hooks/transactions/useSignMultipleTransactions.tsx [hunk 1]
```diff
@@ -109,9 +110,15 @@ export function useSignMultipleTransactions({
     const { tokenId } = transactionTokenInfo;
     const receiver = transaction.getReceiver().toString();
     const sender = transaction.getSender().toString();
-    const isLoggedInWithDifferentAccount = sender && address !== sender;
-
-    // Prevent signing transactions with different account
+    const senderAccount = await getAccount(sender);
+    const isLoggedInWithDifferentAccount =
```
Comment (CiprianDraghici) [inline src/hooks/transactions/useSignMultipleTransactions.tsx:114]: could be extracted and used also on `useSignTransactions.tsx` file at line: 264.

File: src/hooks/transactions/useSignMultipleTransactions.tsx [hunk 2]
```diff
@@ -109,9 +110,15 @@ export function useSignMultipleTransactions({
...
+      senderAccount &&
```
Comment (CiprianDraghici) [inline src/hooks/transactions/useSignMultipleTransactions.tsx:115]: would be very useful to have some tests on this new behavior

File: src/hooks/transactions/useSignTransactions.tsx [hunk 3]
```diff
@@ -259,12 +261,26 @@ export const useSignTransactions = () => {
       return;
     }
 
-    const isLoggedInWithDifferentAccount = transactions.some((tx) => {
-      const sender = tx.getSender().toString();
-      return sender && address !== sender;
-    });
-
-    // Prevent signing transactions with different account
+    const senderAddresses = uniq(
```
Comment (arhtudormorar) [inline src/hooks/transactions/useSignTransactions.tsx:264]: you could throw an error here if one of the senders is different. So if you consider only one sender, then you only fetch info about one account

**Review by CiprianDraghici — APPROVED** [review]

File: src/hooks/transactions/helpers/checkIsInvalidSender.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { AccountType } from 'types/account.types';
+
+export const checkIsInvalidSender = (
```
Comment (arhtudormorar) [inline src/hooks/transactions/helpers/checkIsInvalidSender.ts:3]: checkIsValidSender (it's easyer to work with positive concepts)

File: src/hooks/transactions/useSignMultipleTransactions.tsx [hunk 5]
```diff
@@ -109,9 +113,15 @@ export function useSignMultipleTransactions({
     const { tokenId } = transactionTokenInfo;
     const receiver = transaction.getReceiver().toString();
     const sender = transaction.getSender().toString();
-    const isLoggedInWithDifferentAccount = sender && address !== sender;
+    const senderAccount = await getAccount(sender);
+    const isLoggedInWithDifferentAccount = checkIsInvalidSender(
```
Comment (arhtudormorar) [inline src/hooks/transactions/useSignMultipleTransactions.tsx:117]: isSenderValid

File: src/hooks/transactions/useSignMultipleTransactions.tsx [hunk 6]
```diff
@@ -109,9 +113,15 @@ export function useSignMultipleTransactions({
     const { tokenId } = transactionTokenInfo;
     const receiver = transaction.getReceiver().toString();
     const sender = transaction.getSender().toString();
-    const isLoggedInWithDifferentAccount = sender && address !== sender;
+    const senderAccount = await getAccount(sender);
```
Comment (arhtudormorar) [inline src/hooks/transactions/useSignMultipleTransactions.tsx:116]: try catch or is it caught inside?

File: src/hooks/transactions/useSignMultipleTransactions.tsx [hunk 7]
```diff
@@ -109,9 +113,15 @@ export function useSignMultipleTransactions({
...
+    const isLoggedInWithDifferentAccount = checkIsInvalidSender(
+      senderAccount,
+      address
+    );
 
-    // Prevent signing transactions with different account
+    // Don't allow signing if the sender from the transaction
```
Comment (arhtudormorar) [inline src/hooks/transactions/useSignMultipleTransactions.tsx:122]: Don't allow signing if the logged in account's address is neither the sender or the sender account's active guardian

File: src/hooks/transactions/useSignTransactions.tsx [hunk 8]
```diff
@@ -259,12 +262,28 @@ export const useSignTransactions = () => {
...
+
+    if (senderAddresses.length > 1) {
+      throw new Error('Multiple senders are not allowed');
+    }
+
+    const senderAccount = senderAddresses.length
+      ? await getAccount(senderAddresses[0])
+      : null;
+
+    const isLoggedInWithDifferentAccount = checkIsInvalidSender(
```
Comment (arhtudormorar) [inline src/hooks/transactions/useSignTransactions.tsx:279]: isSenderValid

File: src/hooks/transactions/useSignTransactions.tsx [hunk 9]
```diff
@@ -259,12 +262,28 @@ export const useSignTransactions = () => {
...
+      senderAccount,
+      address
+    );
 
-    // Prevent signing transactions with different account
+    // Don't allow signing if the sender from the transaction
```
Comment (arhtudormorar) [inline src/hooks/transactions/useSignTransactions.tsx:284]: seems repetitive. Maybe only add the comment inside checkIsValidSender ?

File: src/hooks/transactions/useSignMultipleTransactions.tsx (hunk 6 above)
Comment (razvantomegea) [reply src/hooks/transactions/useSignMultipleTransactions.tsx:116]: Caught inside

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**